### ISSUE TITLE
Generic interfaces do not support nested interfaces

### ISSUE THREAD

**Issue by enagy27** [reactiveui/refit#210]
Trying to use Generic Interfaces for non-homogenized endpoints, where one implements a read-only interface (get-only operations), and another implements a read-write interface (all verbs implemented), devolves into the following pattern:
- Break the interfaces down into the separate generic CRUD verb definitions
- Implement a client for each endpoint (i.e. /users), which contains a reference to a rest service for each of the verbs it supports
- Package each of these clients into a parent class which then encompasses the entire API

Unfortunately this pattern makes the calling convention for something as simple as a get: myClient.Users.Getter.Get(someGuid).

Even if I were to create clients where there is no chance of name overlap, however, this is still not possible because Refit requires attribution at the current level. This makes it impossible to define my clients for the calling convention: myClient.Users.Get(someGuid). At this point, this starts to seem a bit restrictive because my design is forced into a "has a" relationship where I wanted to define an "is a" relationship.

**Comment by ahmedalejo** [comment]
@airwareic try adding non-confidential code snippets that reproduces the issue you´re facing.

**Comment by airwareic** [comment]
Sure thing! My original comment contained some stuff that was a bit nutty and didn't make type-sense. I think it's likely that I'm misguided in thinking this should/would work in a framework without throwing convention out the window, but I figured I'd bring it up none the less. Here's what I'm trying to do:
```
public interface IGet<T>
{
    [Get("")]
    Task<IEnumerable<T>> Get();

    [Get("/{id}")]
    Task<IEnumerable<T>> GetSpecific(Guid id);
}

public interface IPost<T>
{
    [Post("")]
    Task Post([Body] T data);
}
```
This is what I tried to do next:
`public interface IUserService : IGet<IUser>, IPost<IUser>`
This does not work because generation will only apply directly to methods with Refit attributes, as opposed to looking at all types which implement the attributed method and duplicating the implementation. Perhaps this would require the addition of a type-level attribute to indicate that an interface should also be stubbed.
